blogposts/2021/continuous_delivery.md

```diff
@@ -0,0 +1,174 @@
+
+Continuity is the dream: if deadlines cause anxiety, then why not get rid of deadlines? By releasing software continuously, you both ensure each code change can make it in and ensure each code change will have the time to get tested and inspected.
+
+And yet, there are limitations.
+
+### Pursuing scalable deployment
+
+Continuous deployment is not a new idea. In a [2009 essay](http://timothyfitz.com/2009/02/08/continuous-deployment/)–in one of the first instances of the concept–Timothy Fitz introduced the idea with an all too recognizable narrative:
+
+> _Alex has just written a refactoring of some website backend code. Since it was a small task, it’s committed and Alex moves on to the next feature._
+
+> _When the code is deployed in production two weeks later it causes the entire site to go down. A one-character typo which was missed by automated tests caused a failure cascade reminiscent of the bad-old-days at twitter. It takes eight hours of downtime to isolate the problem, produce a one character fix, deploy it and bring production back up._
+
+> _Alex curses luck, blames human infallibility, inevitable cost of software engineering and moves on to the next task._
+
+> _This story is the day-to-day of most startups I know. It sucks. Alex has a problem and she doesn’t even know it. Her development practices are unsustainable. “Stupid mistakes” like the one she made happen more frequently as the product grows more complex and as the team gets larger. Alex needs to switch to a **scalable** solution._ (Emphasis added).
+
+The purpose of continuous deployment, then, in Fitz’s definition and the current definition, is to make deployment scalable.
+
+Fitz was perhaps more right than he realized. Traditional deployment practices have only gotten less practical and less scalable.
+
+In our research (see the [Big code report](https://info.sourcegraph.com/emergence-of-big-code-2020-survey?_ga=2.117030534.1813537383.1623859682-2046487521.1621427028)), we found that 51% of respondents reported having more than 100 times the volume of code they had 10 years ago and 18% saying they have 500 times more code.
+
+As Fitz pointed out 12 years ago, so-called “stupid mistakes [...] happen more frequently as the product grows more complex.” But they’re _not_ stupid mistakes, we need to point out; if something is inevitable and consistent, it’s not an individual mistake–it’s a flaw in the system.
+
+That’s why Fitz proposed continuous deployment as a new system. If you’re deploying continuously, he writes, then you can fail fast. He writes that “the closer a failure is to the point where it was introduced, the more data you have to correct for that failure.”
+
+If Alex had deployed as soon as she developed the change, the failure would have been less severe, the rollback faster, and the diagnosis easier to make. And in a puff of smoke, theoretically, goes release anxiety.
+
+### True continuous deployment is impossible for all
```

Comment:
I wonder if you can condense this section a bit? It feels like spending a lot of time validating that on-prem is still a big thing, but as the reader I'm impatient to get to where you acknowledge how this is incompatible with CD 😆 I imagine you are addressing people who work on on-prem software because they are ones who would be interested in the solutions you offer, so they won't need much convincing.